src/python/WMCore/Services/DBS/DBS3Reader.py

```diff
@@ -916,29 +924,11 @@ def fixMissingParentageDatasets(self, childDataset, insertFlag=True):
         self.logger.info("Found %d blocks without parentage information", len(blocks))
         for blockName in blocks:
```

Comment:
this is sequential logic which will grow significantly with number of processed blocks. My advise to make it concurrent and process blocks in parallel, but it will require changing logic to use `async` or rely on queue and multiprocessing modules.

Reply:
This is beyond the scope of these changes, but it can definitely be changed at some point. Feel free to open a GH issue for that.